src/kzg10/fastpoly.rs

```diff
@@ -0,0 +1,372 @@
+use crate::Error;
+use ark_ec::PairingEngine;
+use ark_ff::{Field, One, Zero};
+use ark_poly::UVPolynomial;
+
+use ark_poly::polynomial::univariate::DensePolynomial;
+
+type P<E: PairingEngine> = DensePolynomial<E::Fr>;
+
+/// Computes the inverse of f mod x^l
+pub fn inverse_mod_xl<E: PairingEngine>(f: &P<E>, l: usize) -> Option<P<E>> {
```

Comment:
Could you add a reference/some comments about what the algorithm is here?

Reply:
It's algorithm 9.3 in Modern Computer Algebra by Gathen and Gerhard, but yeah I'll add some references for all the algorithms.